packages/cozy-doctypes/src/File.js

```diff
@@ -23,6 +23,52 @@ class CozyFile extends Document {
     const parentDirectoryPath = trimEnd(parentDir.path, '/')
     return `${parentDirectoryPath}/${name}`
   }
+
+  /**
+   * Move file to destination.
+   *
+   * @param   {string} fileId               - The file's id (required)
+   * @param   {object} destination
+   * @param   {string} destination.folderId - The destination folder's id (required)
+   * @param   {string} destination.path     - The file's path after the move (optional)
+   * @param   {string} force                - Whether we should overwrite the destination in case of conflict (defaults to false)
+   * @returns {Promise}                     - A promise that returns the move action response and the deleted file id (if any) if resolved or an Error if rejected
+   *
+   */
+  static async move(fileId, destination, force = false) {
+    const filesCollection = this.cozyClient.collection('io.cozy.files')
+    try {
+      const resp = await filesCollection.updateFileMetadata(fileId, {
+        dir_id: destination.folderId
+      })
+
+      return {
+        moved: resp.data,
+        deleted: null
+      }
+    } catch (e) {
+      if (e.status === 409 && force) {
+        if (!destination.path || typeof destination.path !== 'string') {
+          throw new Error(
+            'You must provide a destination path if you use force option.'
+          )
+        }
+        const conflictResp = await filesCollection.statByPath(destination.path)
+        await filesCollection.destroy(conflictResp.data)
+
+        const resp = await filesCollection.updateFileMetadata(fileId, {
+          dir_id: destination.folderId
+        })
```

Comment:
Should we try / catch those 2 calls ?

Reply:
Ou c'est à l'app qui appelle de gérer ça proprement ?

Reply:
Si je demande un move je m'attends plutôt à ce qu'il mette à jour les métadonnées lui-même

Reply:
Je parlais du catch si jamais y a une erreur sur une des 2 requêtes (statByPath / updateFileMeta)

Reply:
Oh, dans ce cas je dirais que sauf si tu as quelque chose de particulier à faire lors de l'erreur (genre une clause finally, ou une récupération), je préfère faire les catch le plus haut possible dans l'appelant. Les erreurs ignorées parce que catchées trop basses je trouve ça dangereux

Reply:
What would do if there is an error?